src/rdfTypes/GeometryInfo.h

```diff
@@ -84,18 +85,29 @@ struct GeometryType {
   QL_DEFINE_DEFAULTED_EQUALITY_OPERATOR_LOCAL_CONSTEXPR(GeometryType, type_)
 };
 
+// Represents the length of the geometry in meters.
+struct MetricLength {
+ private:
+  double length_;
+
+ public:
+  MetricLength(double length);
+
+  double length() const { return length_; }
+};
+
```

Comment:
For such simple strong types, you could use the `TypedIndex` class template to reduce the boilerplate code.